sig-and-wg/sig-core/proposals/modularization.md

```diff
@@ -0,0 +1,72 @@
+# Kyma modularization - install components as addons
+
+Created on 2018-08-16 by Piotr Kopczynski (@PK85) and Piotr Bochyński (@pbochynski).
+
+## Status
+
+Proposed on 2018-08-16.
+
+## Motivation
+
+Kyma provides components as a one big pack to enable developers to build something on cloud native stack. They work each other well and together expose unified developer and business experience. Now, we need to go one step back and enable users to decide what they want to use from Kyma.
+
+The landscape around Kyma has changed. Kubernetes managed clusters get popular. Now each big cloud infrastructure provider has own managed kubernetes service (Google GKE, Amazon EKS, Microsoft AKS). This offering is evolving and customers can use more services included in the managed kubernetes (like Istio - service mesh, service catalog, logging). Developers work on bringing more features to transform kubernetes from container scheduling system into complete platform solution. Kyma is not the only solution in this area. Knative is the new open source project started by Google that also aims to provide components to deliver cloud native software on top of kubernetes. There is strategic decision to run Kyma on top of knative. All this development in the kubernetes landscape shows the need of having flexible loosely coupled components to enable customers to use best available solutions from infrastructure providers, open source projects and commercial solutions. Kyma in the principle offers complete solution (batteries included), but it doesn't mean Kyma can't benefit of running on top of managed cluster, by integration with already provided services.
+
+## Goal
+
+Allow users to add/remove/replace Kyma components in more flexible way during the installation and in runtime. This should be possible:
+
+- install Kyma lite on minikube (without monitoring, console and ui-api-layer, tracing, service catalog, and remote environments)
+- install Kyma without service catalog and brokers installed on GKE with already provided Service Catalog and GCP broker included
+- install Kyma as gardener addon Gardener with dependencies to Istio and Knative addons
+- add service broker to existing Kyma installation in runtime (e.g. azure broker)
```

Comment:
`a Service Broker`
~(e.g. azure broker)~ `, for example Azure Broker`

Reply:
done